src/packet/header/fragment.rs

```diff
@@ -0,0 +1,90 @@
+use super::{HeaderParser, HeaderReader};
+use super::PacketHeader;
+use error::{Result, NetworkError};
+
+use std::io::{self, Cursor, Error, ErrorKind, Write};
+use byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
+
+#[derive(Copy, Clone, Debug)]
+/// This header represents an fragmented packet header.
+pub struct FragmentHeader {
+    pub sequence: u16,
+    pub id: u8,
+    pub num_fragments: u8,
+    pub packet_header: Option<PacketHeader>,
+}
+
+impl FragmentHeader {
+    /// Create new fragment with the given packet header
+    pub fn new(id: u8, num_fragments: u8, packet_header: PacketHeader) -> Self {
+        FragmentHeader { id, num_fragments, packet_header: Some(packet_header), sequence: packet_header.seq }
+    }
+
+    /// Get the size of this header.
+    pub fn size(&self) -> u8
+    {
+        if self.id == 0 {
+            if self.packet_header.is_some() {
+                return self.packet_header.as_ref().unwrap().size() + 5;
+            }else {
+                error!("Attemtping to retrieve size on a 0 ID packet with no packet header");
+                0
+            }
+        } else {
+            5
```

Comment:
Could you add a comment here why you just have 5 hard coded?

Reply:
Because 5 is a nice number 😉